### PR TITLE
Feature/cdap 1247 set stream format cli

### PR DESCRIPTION
**Pull request by albertshau**

### CODE REVIEW

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String streamId = arguments.get(ArgumentName.STREAM.toString());
+    StreamProperties currentProperties = streamClient.getConfig(streamId);
+
+    String formatName = arguments.get(ArgumentName.FORMAT.toString());
+    Schema schema = getSchema(arguments);
+    Map<String, String> settings = Collections.emptyMap();
+    if (arguments.hasArgument(ArgumentName.SETTINGS.toString())) {
+      settings = Splitter.on(" ").withKeyValueSeparator("=").split(arguments.get(ArgumentName.SETTINGS.toString()));
```
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java]: Put in some common code which parses key-value input, possibly in `AbstractCommand`. Would be nice to use the same code to parse all key-value input in the CLI, but this can probably be done in another PR.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("set stream properties <%s> <%s>", ArgumentName.STREAM, ArgumentName.LOCAL_FILE_PATH);
+  }
+
+  @Override
+  public String getDescription() {
+    return "Sets the properties of a stream, such as TTL, format, and notification threshold.";
```
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: Is it possible to set all of these individually?

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+      reader.close();
+    }
+
+    streamClient.setStreamProperties(streamId, streamProperties);
+    output.printf("Successfully set properties of stream '%s'\n", streamId);
+  }
+
+  @Override
+  public String getPattern() {
+    return String.format("set stream properties <%s> <%s>", ArgumentName.STREAM, ArgumentName.LOCAL_FILE_PATH);
```
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: Maybe "import stream config <stream-id> <config-file>"? Not sure, just a suggestion.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java (hunk 1 above)
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamFormatCommand.java]: good point, opened CDAP-1267

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java (hunk 2 above)
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: they are all set through the same API ```PUT streams/<id>/config```, though only specifying one will only change that setting. For example, only specifying ttl will only change ttl.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java (hunk 3 above)
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: yeah wasn't sure what syntax we should use. Seems like the other commands that reference files start with "load ..." but that seems very close to the syntax for loading a file to the stream

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java (hunk 2 above)
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: Makes sense. Do we also have a way to do this via text input rather than file?

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java (hunk 3 above)
Comment (jwang47) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: Actually I think "set stream properties" is fine for now. Later I'll revise the command format style to make it easier to use though.

File: cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java (hunk 2 above)
Comment (albertshau) [inline cdap-cli/src/main/java/co/cask/cdap/cli/command/SetStreamPropertiesCommand.java]: there is one command for ttl and one command for format/schema. I'm not really sure what threshold is for beyond that its related to notifications, so leaving that for later.

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-RBT99

**Comment by jwang47** [comment]
LGTM
